### ISSUE TITLE
Handling common code type declarations

### ISSUE THREAD

**Issue by mbarlow12** [phetsims/eliot#2]
We can specify specific places to look for type definitions (`.d.ts` files) in `tsconfig.json`. However, this would involve either copying the same definitions in each repo or hardcoding the various paths of type declarations in `tsconfig.json`.

Like the requirejs config files, we can define module paths in `tsconfig` that are aliased to more convenient strings (e.g. PHET_CORE instead of '../../phet-core/js/...'). What would be ideal (I think) would be a way to reference a `types` directory relative to the above alias. In this way, our types path would simply be `../types` and all types in a given repo would be in the top level.

A final option is to simply have type declarations alongside their corresponding .js files, so in scenery-phet you'd see:
```js
...
ResetAllButton.js
ResetAllButton.d.ts
...
```
While this works quite seamlessly and is pretty clear, I would be very hesitant to double the number of files in all our repos.

**Comment by samreid** [comment]
We only need *.d.ts files for *.js files that are not ported to *.ts, right?

**Comment by mbarlow12** [comment]
Correct, this is how new .ts files can recognize the API from linked .js files.
